pyirf/io/io.py

```diff
@@ -49,40 +87,179 @@ def read_simu_info_merged_hdf5(filename):
+            "obs_id",
+        ]
         for k in colnames:
             if k not in not_to_check:
                 assert np.all(simu_info[:][k] == simu_info[0][k])
-        num_showers = simu_info[:]['num_showers'].sum()
+        num_showers = simu_info[:]["num_showers"].sum()
 
     combined_mcheader = read_simu_info_hdf5(filename)
-    combined_mcheader['num_showers'] = num_showers
+    combined_mcheader["num_showers"] = num_showers
     return combined_mcheader
 
 
 def get_simu_info(filepath, particle_name, config={}):
-    """
-    read simu info from file and return config
-    """
+    """Read simulated information from file and return config."""
 
-    if 'particle_information' not in config:
-        config['particle_information'] = {}
-    if particle_name not in config['particle_information']:
-        config['particle_information'][particle_name] = {}
-    cfg = config['particle_information'][particle_name]
+    if "particle_information" not in config:
+        config["particle_information"] = {}
+    if particle_name not in config["particle_information"]:
+        config["particle_information"][particle_name] = {}
+    cfg = config["particle_information"][particle_name]
 
     simu = read_simu_info_merged_hdf5(filepath)
-    cfg['n_events_per_file'] = simu.num_showers * simu.shower_reuse
-    cfg['n_files'] = 1
-    cfg['e_min'] = simu.energy_range_min
-    cfg['e_max'] = simu.energy_range_max
-    cfg['gen_radius'] = simu.max_scatter_range
-    cfg['diff_cone'] = simu.max_viewcone_radius
-    cfg['gen_gamma'] = -simu.spectral_index
+    cfg["n_events_per_file"] = simu.num_showers * simu.shower_reuse
+    cfg["n_files"] = 1
+    cfg["e_min"] = simu.energy_range_min
+    cfg["e_max"] = simu.energy_range_max
+    cfg["gen_radius"] = simu.max_scatter_range
+    cfg["diff_cone"] = simu.max_viewcone_radius
+    cfg["gen_gamma"] = -simu.spectral_index
 
     print(particle_name)
     print(cfg)
 
     return config
+
+
+def GADF_mapper(debug=False, config=None):
+    """Defines the format to be used internally after input.
+
+    It should be always based on the latest version of [1]_.
+    All readers should call this function to map input data from different
+    formats.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary obtained from pyirf.io.load_config.
+    debug  : bool
+        If True, print some debugging information.
+
+    Returns
+    -------
+
+    columns : dict
+        Dictionary that maps user-defined DL2 quantities to the GADF equivalent.
+
+    Notes
+    -----
+
+    .. [1] https://gamma-astro-data-formats.readthedocs.io/en/latest/
+
+    """
+
+    columns = {}
+
+    # columns["GADF_DEF"] = config["column_definition"]["USER DEF"]
+
+    for key in config["column_definition"]:
+        columns[key] = config["column_definition"][key]
+
+    if debug:
+        print("MAPPING TO GADF....")
+        print(columns)
+
+    return columns
+
+
+def read_FITS(config=None, infile=None, debug=False):
```

Comment:
As discussed elsewhere, there simply is no DL2 GADF data format.

I cannot stress this enough: we need to define our internal data format and then people can use the library for their data by putting it into those data structures.

Reply:
ouch , OK, I didn't get this problem fully before...

I guess that then this function is still doing it's job somehow (clearly, since the script works), but the context is wrong.

We can check and define the internal data format in a subsequent PR, but for this PR I would suggest then 

- to change function name (something like internal_dataformat_mapper) and its docstring to something that reflects this change in scope
- perhaps remove the check for the exitence of columns which then shouldn't be at all in DL2 simulated data

Reply:
also, I think I'd like to change `read_FITS` into `read_EventDisplay`, since the FITS HDUs structur is specific to that pipeline

Reply:
and in the config file, for the reasons you explained we should then remove only TIME, RA and DEC, correct?

Reply:
Neither GLON/GLAT nor TIME/RA/DEC will make sense for any of the current simulated events.